src/Data/Singletons/Prelude/Foldable.hs

```diff
@@ -370,26 +360,28 @@ $(singletonsOnly [d|
 
       -- -| The largest element of a non-empty structure.
       maximum :: forall a . Ord a => t a -> a
-      maximum x =
-        case foldMap (MaxInternal . Just) x of
-          MaxInternal y -> fromMaybe (errorWithoutStackTrace "maximum: empty structure") y
+      maximum = fromMaybe (errorWithoutStackTrace "maximum: empty structure") .
```

Comment:
Ideally, I would have written this:

```hs
      maximum = fromMaybe (errorWithoutStackTrace "maximum: empty structure") .
       getMaxInternal . foldMap (MaxInternal . (Just :: a -> Maybe))
```

As that is closer to how `maximum` is actually defined in `base`. Sadly, this won't work in `singletons` due to #433.